iroha/src/account.rs

```diff
@@ -293,25 +263,61 @@ pub mod isi {
     /// #
     /// # let command_payload = &CreateRole {
     /// #    role_name: "user".to_string(),
-    /// #    permissions: Vec::new(),
+    /// #    permissions: vec!["buy".to_string()],
     /// # };
     /// # let result: Vec<u8> = command_payload.into();
     /// let command_payload: CreateRole  = result.into();
     /// ```
     impl std::convert::From<Vec<u8>> for CreateRole {
         fn from(command_payload: Vec<u8>) -> Self {
-            bincode::deserialize(&command_payload).expect("Failed to deserialize payload.")
+            <CreateRole>::decode(&mut command_payload.as_slice())
+                .expect("Failed to deserialize payload.")
         }
     }
 
-    #[test]
-    fn create_role_command_serialization_and_deserialization() {
-        let expected = CreateRole {
-            role_name: "user".to_string(),
-            permissions: Vec::new(),
-        };
-        let actual: CreateRole =
-            bincode::deserialize(&bincode::serialize(&expected).unwrap()[..]).unwrap();
-        assert_eq!(expected, actual);
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+
+        #[test]
+        fn append_role_command_serialization_and_deserialization() {
+            let expected = AppendRole {
+                account_id: Id::new("account", "domain"),
+                role_name: "role".to_string(),
+            };
+            let actual = <AppendRole>::decode(&mut expected.encode().as_slice()).unwrap();
```

Comment:
Does `AppendRole` need to be in `<..>`? I believe it would work without them, simply like `AppendRole::decode(...)`.

Reply:
Same for other test examples.

Reply:
Both options work, IDK why they use `<..>` notation in docs

Reply:
I believe they use them in docs when specifying generic types, at least that's what I got from there examples.